Docs - Develop - EVM (#1245)
iotaledger/wasp: documentation/docs/guide/evm/compatibility.md

- compatibility
- fees
- reference

---

# EVM Compatibility in IOTA Smart Contracts

The [`evm`](../core_concepts/core_contracts/evm.md) [core contract](../core_concepts/core_contracts/overview.md)

> provides EVM support in IOTA Smart Contracts. Its main purpose is to store the EVM state (account balances, state, code,
> etc.) and to provide a way to execute EVM code to manipulate the state.

Reviewer: ```suggestion
provides EVM support in IOTA Smart Contracts. It stores the EVM state (account balances, state, code,
etc.) and provides a way to execute EVM code to manipulate the state.
```

The EVM core contract runs on top of the ISC layer, which provides the rest of the machinery needed to run smart
contracts, such as signed requests, blocks, state, proofs, etc.

However, the ISC EVM layer is also designed to be as compatible as possible with existing Ethereum tools

> like [MetaMask](https://metamask.io/), which assume that the EVM code will be run on an Ethereum blockchain composed of

Reviewer: ```suggestion
like [MetaMask](https://metamask.io/), which assume that the EVM code runs on an Ethereum blockchain composed of
```

> Ethereum blocks containing Ethereum transactions. Unfortunately, since ISC works in a fundamentally different way,

Reviewer: ```suggestion
Ethereum blocks containing Ethereum transactions. Since ISC works in a fundamentally different way,
```

providing 100% compatibility is not possible. We do our best to emulate the behavior of an Ethereum node, so the
Ethereum tools think they are interfacing with an actual Ethereum node, but some differences in behavior are inevitable.

Here are some of the most important properties and limitations of EVM support in IOTA Smart Contracts:

> The Wasp node provides a JSON-RPC service, the standard protocol used by Ethereum tools. Upon receiving a signed

Reviewer: A bullet is missing. Would it make sense to introduce a header and turn bullet points into normal paragraphs?
